### PR TITLE
feat(components): Docs Site Side Nav Links

### PR DESCRIPTION
**Pull request by scotttjob**
## Motivations

1. We wanted to add links to various sections within our documents.
2. So we've added a bit of code that adds some markers to any MDX processed H2 headers, so we can snag them in the links section.
3. We needed an addition to our tabs component to make them controlled, so we could take over and switch tabs if the user clicked on a link that was on a different tab.

## Changes

1. Added the ability to dynamically link to sections in our MDX files.

## Testing

Check out any component page. You should see a few new links in the top right hand corner, some corresponding to H2s in the Design tab, and then two for Usage/Props in the Code tab.

image

Changes can be
[tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/site/src/components/ComponentLinks.tsx [hunk 1]
```diff
@@ -9,16 +10,68 @@ import { useAtlantisSite } from "../providers/AtlantisSiteProvider";
...
+  };
   const { isMinimal } = useAtlantisSite();
   if (isMinimal) return null;
 
   return (
     <Content>
       <Box padding="base" direction="column">
-        <Heading level={2}>Links</Heading>
+        <Box margin={{ bottom: "base" }}>
+          <Heading level={3}>Design</Heading>
```
Comment (chris-at-jobber) [inline packages/site/src/components/ComponentLinks.tsx:54]: [To match Figma](https://www.figma.com/design/HhNrPoreO9D8R8O9Ta2AM7/Unified-Atlantis-Docs?node-id=1333-20309&t=TCOMi34VXaWhjAec-1), should be using 6
```suggestion
          <Heading level={6} element={h3}>Design</Heading>
```

File: packages/site/src/components/ComponentLinks.tsx [hunk 2]
```diff
@@ -9,16 +10,68 @@ import { useAtlantisSite } from "../providers/AtlantisSiteProvider";
...
+          {hlinks?.map((link, index) => (
+            <Box key={index}>
+              <a onClick={click} href={`#${link.id}`}>
+                {link.textContent}
+              </a>
+            </Box>
+          ))}
+        </Box>
+        <Box margin={{ bottom: "base" }}>
```
Comment (chris-at-jobber) [inline packages/site/src/components/ComponentLinks.tsx:63]: I don't think these `Box`es need a margin if we wrap the lot of em in a `Content`

Comment (chris-at-jobber) [reply packages/site/src/components/ComponentLinks.tsx:63]: Overall structure is 

`base` spacing (ie default `Content`)between headers and links in a section

`larger` spacing (ie `Content spacing="larger"`) between sections

smaller spacing between Icon and Link in the "external" Links section (maybe `Flex direction="horizontal" template={["shrink", "grow"]} gap="smaller"`?)

Comment (chris-at-jobber) [reply packages/site/src/components/ComponentLinks.tsx:63]: For the spacing between Links, looks like we might only need to add `smaller` (Figma in red, screenshot of rendered in green)
image

File: packages/site/src/components/ComponentLinks.tsx [hunk 3]
```diff
@@ -9,16 +10,68 @@ import { useAtlantisSite } from "../providers/AtlantisSiteProvider";
...
+            </a>
+            <a onClick={goToProps} href={`#`}>
+              Props
+            </a>
+          </Box>
+        </Box>
+        <Heading level={3}>Links</Heading>
         <Box>
           {links?.map((link, index) => (
             <Box key={index}>
```
Comment (chris-at-jobber) [inline packages/site/src/components/ComponentLinks.tsx:77]: These external links should also have the Link icon:
image

**Comment by taylorvnoj** [comment]
Hey @scotttjob - I'm still testing but seems like the links for Usage and Props are not behaving for me. I did an `npm install` and `npm run dev` to start the branch fresh - are these links working for you?



https://github.com/user-attachments/assets/621d85a6-cb1b-49f6-9c1e-1c6c4d24f0ab

**Review by taylorvnoj — COMMENTED** [review]
Maybe let's go through this together on Monday, I think I'm missing context possibly.

I'm also wondering if, when components have both Web and Mobile, the Usage and Props links should have mobile/web distinction? That way if I'm on the Design tab and click Props - Mobile (or something) I can go straight to mobile props

**Comment by scotttjob** [comment]
> Maybe let's go through this together on Monday, I think I'm missing context possibly.
> 
> I'm also wondering if, when components have both Web and Mobile, the Usage and Props links should have mobile/web distinction? That way if I'm on the Design tab and click `Props - Mobile` (or something) I can go straight to mobile props

This is a good callout, I think I started this link work before there was a mobile tab.

File: packages/site/src/components/ComponentLinks.tsx (hunk 2 above)
Comment (scotttjob) [reply packages/site/src/components/ComponentLinks.tsx]: I THINK I've addressed this, but I didn't compare it back to the Figma. So let's sit down for 10 if it's still off and we'll adjust together until it's good to go.

File: packages/site/src/components/ComponentLinks.tsx (hunk 3 above)
Comment (scotttjob) [reply packages/site/src/components/ComponentLinks.tsx]: Done!

**Comment by scotttjob** [comment]
> Maybe let's go through this together on Monday, I think I'm missing context possibly.
> 
> I'm also wondering if, when components have both Web and Mobile, the Usage and Props links should have mobile/web distinction? That way if I'm on the Design tab and click `Props - Mobile` (or something) I can go straight to mobile props

I've updated this PR to support both "Web" and "Mobile"

**Review by taylorvnoj — APPROVED** [review]
LGTM! Side nav links are going where they should in the documentation and if I'm on Design Tab and click the props link, I go there & vice versa! Amazing stuff!
🚢

**Comment by taylorvnoj** [comment]
@scotttjob I pulled down the changes - approval still stands! The previous noted issue is fixed!

**Pull request merged**